sentry/src/main/java/io/sentry/SentryThreadFactory.java

```diff
@@ -90,11 +90,12 @@ List<SentryThread> getCurrentThreads(
       for (Map.Entry<Thread, StackTraceElement[]> item : threads.entrySet()) {
 
         final Thread thread = item.getKey();
+        final boolean isCurrent = thread == currentThread;
         final boolean crashed =
-            (thread == currentThread)
+            isCurrent
                 || (mechanismThreadIds != null && mechanismThreadIds.contains(thread.getId()));
 
-        result.add(getSentryThread(crashed, item.getValue(), item.getKey()));
+        result.add(getSentryThread(isCurrent, crashed, item.getValue(), item.getKey()));
```

Comment:
https://develop.sentry.dev/sdk/event-payloads/threads/#attributes
It's a bit confusing but here we go:
`crashed` means, this was the thread that caused the error, or captured the error if best-effort, that's why we double-check the thread-id or if its the current thread.
`current` means, if the thread was in the foreground, I'm not entirely sure how to determine that, if we can use https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.State.html or something.
I'd double-check the `current` one with folks on client-infra since I'm not 100% sure right now.

This is not set here because on Android, we set ourselves on the Android event processor and right now it's only setting because it's null.
If we want to set this flag for Java, we'd need to overwrite the value on Android, so be careful with this change.

Reply:
So if it's not needed and we can't merge as is. Do we just close it and revisit later if it becomes relevant again?

Reply:
It's not part of v6 so don't feel blocked by it, but I'd double-check with folks the meaning of the `current` flag.
Android sets the `current` flag if it's the UI/main thread and I believe it's wrong as well, so I guess we can keep it open and fix it for both SDKs, we just need to be sure about the correct intention of this flag.